apps/prairielearn/src/pages/workspace/workspace.js

```diff
@@ -16,32 +17,55 @@ const { Workspace } = require('./workspace.html');
 
 const sql = sqldb.loadSqlEquiv(__filename);
 
-router.get('/', (_req, res, _next) => {
-  let navTitle;
+async function getNavTitleHref(res) {
   if (res.locals.assessment == null) {
-    // instructor preview
-    res.locals.pageNote = 'Preview';
-    res.locals.pageTitle = res.locals.question_qid;
-    navTitle = res.locals.pageTitle;
+    const variant_id = await selectVariantIdForWorkspace(res.locals.workspace_id);
+
+    // Instructor preview. This could be a preview at either the course or course
+    // instance level. Generate a link appropriately.
+    if (res.locals.course_instance_id) {
+      return `/pl/course_instance/${res.locals.course_instance_id}/instructor/question/${res.locals.question_id}/preview?variant_id=${variant_id}`;
+    } else {
+      return `/pl/course/${res.locals.course_id}/question/${res.locals.question_id}/preview?variant_id=${variant_id}`;
+    }
   } else {
-    // student assessment
-    navTitle = `${res.locals.instance_question_info.question_number} - ${res.locals.course.short_name}`;
+    // Student assessment.
+    return `/pl/course_instance/${res.locals.course_instance_id}/instance_question/${res.locals.instance_question_id}`;
```

Comment:
For homeworks the variant ID of the instance question may not be the same as the workspace, particularly if the student has already created a new variant. I believe you'll need to add the variant ID as a query param to avoid confusion in this case. Note that the instance question reads this parameter for homework questions, and ignores it for exam questions:
https://github.com/PrairieLearn/PrairieLearn/blob/0f540bca2fccb6d732eeee53b3dd2d6385a4fc83/apps/prairielearn/src/pages/studentInstanceQuestion/studentInstanceQuestion.js#L242

Reply:
Great point, let me get that fixed up.

Reply:
Done in b9b4772.